### DISCUSSION TITLE
Create a database with comments

### DISCUSSION THREAD

**Discussion by zoeleu** [ThioJoe/YT-Spammer-Purge#325] [Ideas]
This would be useful so other tools can be created with AI or other technologies. Having a dataset of normal/spam comments would be VERY useful.

**Comment by ThioJoe** [comment]
Yea i'm considering adding an option to output the logs in json format, unless you have a better format suggestion

**Reply by Kurdakov** [reply]
I think that json is ok, what he asks a public database. I think - public database might have some problems (if only  commenter names are filtered out (asci parts, but utf phone numbers might be left) and comments ids are replaced with log counter format (comment id 1, 2 ,3 reply to comment 2 etc - for that maybe option - prepare public logs from existing logs might be present ), 
but feature to securely upload public logs somewhere might be good, and seems least problematic - is a public server where links to videos are stored and spam logs are kept private. 

There are actually some spam researches who might at some point be interested to use advanced AI features on logs.
(as there are just few spam databases with limited coverage)
In exchange for logs they could be asked to add those advanced features as options to utility.

but at this point I think there should be set priorities.
current utility is just ok to fight spam (and AI, though could be interesting feature, won't add much)
I think in half year maybe there will be headless version of server, then logs could be collected and offered for AI developers.

**Reply by Kurdakov** [reply]
as for now, reading spam papers seems random forest with n-grams is quite good at youtube spam detection (n-grams are just few symbols (not complete words) - and mostly important what others do not use - your powerful regex features - so n-grams might be in fact some regex expressions such as detecting several cycled utf numbers ), there are quite few random forest python implementations on the net, so maybe some enterprising submitter will add that (possibly even you) to the utility. If it proves that success rate is high, then why not to share collected logs for researches to improve further?

**Reply by Kurdakov** [reply]
btw current logs have no important features for outside analysis:
 for impersonating bots - it's scanned channel and video owner name (while it's possible post process and extract from video link, the information is not explicit) - so analyzing comments for impersonating names/(possibly profile picture, saving to log at least hash of image via ImageHash) outside utility is difficult - but they are exactly important features for algorithm to decide, another feature which is might be important - date of account creation - obviously most bots (especially for newly devised attacks) are recently created ( so date of video + date of bot creation is important for later analysis)

**Reply by zoeleu** [reply]
> 

Everything should be left in! Including names! Thats why a database exists. Many of them have unicode/emoji usernames with phone numbers. It's necessary to leave everything in.

**Reply by zoeleu** [reply]
> Yea i'm considering adding an option to output the logs in json format, unless you have a better format suggestion

A public dataset. This means some sort of API of which the client submits username, video id, comment, and date of submission, then can be exported to JSON or some other format monthly/weekly.

**Reply by ThioJoe** [reply]
> it's scanned channel and video owner name

I should be able to add info like the video title and channel owner name, some of the other stuff like upload date and age of commenter, the app doesn't currently retrieve any of that. I guess if there was enough demand for it I could add a setting to get extra info.

**Reply by ThioJoe** [reply]
@Kurdakov There's a new branch that may interest you https://github.com/ThioJoe/YT-Spammer-Purge/tree/json-extra-data

**Reply by Kurdakov** [reply]
@ThioJoe yes. checking it.
the problem is that if one scans in non channelOwner mode, channelOwnerID is not defined and

def get_extra_json_data(channelIDs, jsonSettingsDict):
  channelOwnerID = jsonSettingsDict['channelOwnerID']

crashes as jsonSettingsDict['channelOwnerID'] is not defined,
then writes obscure message and also writes rtf file 

also it will be fine if rtf plaintext and json were in log_mode and not as additional setting

**Reply by ThioJoe** [reply]
> crashes as jsonSettingsDict['channelOwnerID'] is not defined,
@Kurdakov 
Hm I think I found the problem, did you happen to have json_profile_picture set to false? For me it caused an error but didn't fully crash the program. Could you try the latest commit on that branch and see if it still happens (I've renamed the branch to 2.9.0-Beta)? If not if you could make an issue with stuff like what scan mode, filter mode, etc.

**Reply by Kurdakov** [reply]
@ThioJoe  2.9.0-Beta works ok

**Comment by KendallDoesCoding** [comment]
I had a similar idea

Issue thread can be found here https://github.com/ThioJoe/YT-Spam-Domains-List/issues/44
